crates/nu-parser/src/type_check.rs

```diff
@@ -139,33 +139,35 @@ pub fn math_result_type(
                     )
                 }
             },
-            Operator::Divide | Operator::Modulo => match (&lhs.ty, &rhs.ty) {
-                (Type::Int, Type::Int) => (Type::Int, None),
-                (Type::Float, Type::Int) => (Type::Float, None),
-                (Type::Int, Type::Float) => (Type::Float, None),
-                (Type::Float, Type::Float) => (Type::Float, None),
-                (Type::Filesize, Type::Filesize) => (Type::Float, None),
-                (Type::Duration, Type::Duration) => (Type::Float, None),
+            Operator::Divide | Operator::Modulo | Operator::IntegerDivision => {
+                match (&lhs.ty, &rhs.ty) {
+                    (Type::Int, Type::Int) => (Type::Int, None),
+                    (Type::Float, Type::Int) => (Type::Float, None),
+                    (Type::Int, Type::Float) => (Type::Float, None),
```

Comment:
I don't know if this is ok, for `Operator::IntegerDivision`, the result type should be `(Type::Int, None)`?

But I've tried with some script, but don't get issue about it

Reply:
I think `float // int` could be an integer result but I think `anything // float` should throw an error. IMO integer division doesn't make sense when dividing by a float.

Reply:
Or actually... `6 // 2.5` would result to `2`. That makes sense. I'd make the result an integer result type, though.